frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -136,15 +216,211 @@ export function DynamicConfiguration () {
-                    />
+                    <ExpandableSection 
+                        headerText='Applications'
+                        variant='default' 
+                        headingTagOverride='h3' 
+                        headerDescription='A list of applications for Amazon EMR to install and configure when launching the cluster.'
+                    >
+                        <Multiselect
+                            selectedOptions={selectedApplicationOptions}
+                            onChange={({ detail }) =>
+                                addApplication(detail)
+                            }
+                            options={applicationOptions}
+                            placeholder='Select applications'
+                        />
+                    </ExpandableSection>
+                    <ExpandableSection 
+                        headerText='Cluster Types'
+                        variant='default' 
+                        headingTagOverride='h3' 
+                        headerDescription='The cluster types options that users can select from when creating a new Amazon EMR Cluster.'
+                    >
+                        <ClusterTypeConfiguration
+                            item={
+                                state.form as (IAppConfiguration)
+                            }
+                            setFields={setFields}
+                            touchFields={touchFields}
+                            formErrors={formErrors}
+                        />
+                    </ExpandableSection>
+                    <ExpandableSection 
+                        headerText='Auto Scaling Policy'
+                        variant='default' 
+                        headingTagOverride='h3' 
+                        headerDescription='An automatic scaling policy for a core instance group or task instance group in an Amazon EMR cluster. The automatic scaling policy defines how an instance group dynamically adds and terminates Amazon EC2 instances.'
+                    >
+                        <SpaceBetween direction='vertical' size='m'>
+                            <FormField
+                                label='Max Instances'
+                                constraintText='Must be an integer value.'
+                                errorText={formErrors?.configuration?.EMRConfig?.autoScaling?.maxInstances}
+                                description='The maximum number of instances supporting the Amazon EMR Cluster at any time.'
+                            >
+                                <Input
+                                    data-cy='cluster-max-size'
+                                    value={state.form.configuration.EMRConfig.autoScaling.maxInstances.toString()}
+                                    onChange={(event) => {
+                                        setFields({ 'configuration.EMRConfig.autoScaling.maxInstances': +event.detail.value });
```

Comment:
IMHO I think we should use `Number(event.detail.value)` instead so it is a little more explicit/clear what his happening.

Reply:
Yes but then I don't get to show off my Typescript syntax knowledge.

In the interest of readable code I'll make your change here and everywhere else I use the `+` operator.